client-index/src/service/transaction_service.rs

```diff
@@ -33,13 +42,38 @@ where
     }
 
     /// Sets transaction with given id and value
-    pub fn set(&self, id: &TxId, transaction: &Transaction) -> Result<()> {
-        self.storage.set(KEYSPACE, id, transaction.encode())?;
+    pub fn set(&self, transaction: &Transaction) -> Result<()> {
+        self.storage
+            .set(KEYSPACE, &transaction.id(), transaction.encode())
+            .map(|_| ())
+    }
+
+    /// Retrieves transaction output corresponding to given pointer
+    pub fn get_output(&self, input: &TxoPointer) -> Result<TxOut> {
+        let transaction = self
+            .get(&input.id)?
+            .ok_or_else(|| Error::from(ErrorKind::TransactionNotFound))?;
+
+        let outputs = match transaction {
```

Comment:
I'm thinking whether https://github.com/crypto-com/chain/blob/44d98640c679bd49b7dff13607267c2b8a6f4fad/chain-core/src/tx/mod.rs#L24 could be used to simplify this